content/articles/using-machine-learning-to-translate-brain-signals-to-text/index.md

```diff
@@ -0,0 +1,136 @@
+In the offline system, they used a large-vocabulary language model to simulate an auto-correct feature. It fixed errors by predicting the probability that a letter came next in a sequence of letters. 
+
+### Results
+
+This research shows results for 5 days from one study participant who already had a system implanted in his brain.
+
+To avoid bombarding you with a lot of jargon, I've summarized the results of the finding for you:
+
+1. As the patient continued to do the task of imagining letters and allowing the computer to process these signals repeatedly, the patient was able to type 90 characters per minute for each trial day. This doubled the record compared to previous BCI records for typing. 
+
+2. Since our brain's baseline neural signals change day after day, the handwriting decoder had to be retrained with new data every single day. This retraining had to be accounted for especially when working with models like this. You wouldn't want your training signals to look this way one day and a different way another day. This would make your model not work correctly.
+
+3. The character and word error rates reduced by 0.89% and 3.4% respectively when the offline language model system was used. The result from the word error rate is comparable to state-of-the-art speech recognition systems which range between 4-5%. This is well within the accepted range of being usable.
+
+4. As the participant imagined writing the letters and symbols, the sensors implanted in his brain picked up on patterns of electrical activity which the algorithm interpreted to trace the path of his imaginary. 
+
+![Interpretation](/engineering-education/using-machine-learning-to-translate-brain-signals-to-text/interpretation.PNG)
+
+*[Image Source: HHMI](https://www.hhmi.org/news/brain-computer-interface-turns-mental-handwriting-into-text-on-screen)*
+
+Impressive, right?
+
+5. Finally, this research proved that as much as someone is paralyzed, the brain signals for movement still remain. Tapping into this can help build BCI systems that can help solve a lot of tasks.
+
+Please find on Github the full code for the paper [here](https://github.com/fwillett/handwritingBCI).
+
+### Wrapping Up
+
+As we wrap up, I'd like to point out that this isn't something that will be available to the general public. This is because such a procedure would need implanting electrodes into people's brains. This is something I'm sure many people would mind. It is still in its early experimental phase. Only one patient was used to get these results. More tests still need to be done on more patients to see how well this novel system generalizes. But, the findings obtained from this experiment hold huge promises and would solve a lot of real-world problems. Their biggest hope is that their invention would one day help people with complications such as paralysis. This would enable patients with the disease to able to communicate, both faster and better. 
```

Comment:
```suggestion
As we wrap up, I'd like to point out that this isn't something that will be available to the general public. This is because such a procedure would need implanting electrodes into people's brains. It is still in the experimental phase. Only one patient was used to get these results. More tests still need to be done on more patients to see how well this novel system generalizes. But, the findings obtained from this experiment show huge promise and would have a lot of impactful real-world uses. Their biggest hope is that their invention will one day help people with complications such as paralysis. This would enable patients to communicate, both faster and better. It would also offer them inclusion in the usage of impactful technology, whose purpose is to make lives better.
```